src/azure-cli/azure/cli/command_modules/acr/connected_registry.py

```diff
@@ -0,0 +1,494 @@
+                family_tree[parent_id]["childs"].append(registry.id)
+        result = _get_descendancy(family_tree, root_parent_id)
+    else:
+        result = connected_registry_list
+    return result
+
+
+def acr_connected_registry_show(cmd,
+                                client,
+                                connected_registry_name,
+                                registry_name,
+                                resource_group_name=None):
+    _, resource_group_name = validate_managed_registry(
+        cmd, registry_name, resource_group_name)
+    return client.get(resource_group_name, registry_name, connected_registry_name)
+
+
+def acr_connected_registry_install_info(cmd,
+                                        client,
+                                        connected_registry_name,
+                                        registry_name,
+                                        fresh_install=False,
+                                        resource_group_name=None):
+    registry, resource_group_name = validate_managed_registry(
+        cmd, registry_name, resource_group_name)
+    connected_registry = acr_connected_registry_show(
+        cmd, client, connected_registry_name, registry_name, resource_group_name)
+    connected_registry_localhost = connected_registry.login_server.host
+    parent_gateway_endpoint = connected_registry.parent.sync_properties.gateway_endpoint
+    parent_id = connected_registry.parent.id
+    sync_token_name = connected_registry.parent.sync_properties.token_id.split('/tokens/')[1]
+    if parent_id:
+        parent = parent_id.split('/connectedRegistry/')[1]
+        parent = acr_connected_registry_show(
+            cmd, client, parent, registry_name, resource_group_name)
+        parent_registry_endpoint = parent.login_server.host
+    else:
+        parent_registry_endpoint = registry.login_server
+
+    if fresh_install:
+        from ._client_factory import cf_acr_token_credentials
+        from .token import acr_token_credential_generate
+        cred_client = cf_acr_token_credentials(cmd.cli_ctx)
+        poller = acr_token_credential_generate(
+            cmd, cred_client, registry_name, sync_token_name,
+            password1=True, password2=True, resource_group_name=resource_group_name)
+        #   expiration_in_days=expiration_in_days, expiration=expiration)
+        credentials = LongRunningOperation(cmd.cli_ctx)(poller)
+        sync_username = credentials.username
+        sync_password = {
+            "password1": credentials.passwords[0].value,
+            "password2": credentials.passwords[1].value
+        }
+    else:
+        sync_username = "<sync token username>"
+        sync_password = "<sync token password>"
+
+    return {
+        "ACR_REGISTRY_NAME": connected_registry_name,
+        "ACR_REGISTRY_LOGIN_SERVER": connected_registry_localhost,
+        "ACR_SYNC_TOKEN_NAME": sync_token_name,
+        "ACR_SYNC_TOKEN_USERNAME": sync_username,
```

Comment:
if we give them password, do we need to indicate that it is one time password and should copy? Not sure how the token credentials CLI does this

Reply:
good catch! yes they do. I added a warning